include/hc_printf.hpp

```diff
@@ -82,30 +93,69 @@ static inline void countArg(unsigned int& count, const T& t, const Rest&... rest
   countArg(count,rest...);
 }
 
+static inline unsigned int string_length(const char* str) [[hc,cpu]]{
+  unsigned int size = 0;
+  while(str[size]!='\0')
+    size++;
+  return size;
+}
+
+static inline void copy_n(char* dest, const char* src, unsigned int len) [[hc,cpu]] {
+  for(int i=0; i < len; i++){
+    dest[i] = src[i];
+  }
+}
+
+static inline PrintfError process_str_batch(PrintfPacket* queue, char* fbuffer, int offset, const char* fstring) [[hc,cpu]] {
+  unsigned int fstr_len = string_length(fstring);
+  unsigned int fb_offset = queue[3].data.ai.fetch_add(fstr_len + 1);
+  if (queue[3].data.ui > queue[2].data.ui){
+    return PRINTF_FORMAT_BUFFER_OVERFLOW;
+  }
+  copy_n(&fbuffer[fb_offset], fstring, fstr_len);
+  fbuffer[fb_offset + fstr_len] = '\0';
```

Comment:
It would be betterto do 
  copy_n(&fbuffer[fb_offset], fstring, **fstr_len + 1**);
then explicitly adding an extra '\0' at the end